containers/unit_tests/TestOffsetView.hpp

```diff
@@ -278,6 +278,79 @@ void test_offsetview_construction(unsigned int size) {
 #endif
   }
 }
+
+template <typename Scalar, typename Device>
+void test_offsetview_unmanaged_construction() {
+  // Preallocated memory (Only need a valid address for this test)
+  Scalar s;
+
+  {
+    // Constructing an OffsetView directly around our preallocated memory
+    Kokkos::Array<int64_t, 1> begins1{2};
+    Kokkos::Array<int64_t, 1> ends1{3};
+    Kokkos::Experimental::OffsetView<Scalar*, Device> ov1(&s, begins1, ends1);
+
+    // Constructing an OffsetView around an unmanaged View of our preallocated
+    // memory
+    Kokkos::View<Scalar*, Device> v1(&s, ends1[0] - begins1[0]);
+    Kokkos::Experimental::OffsetView<Scalar*, Device> ovv1(v1, begins1);
+
+    // They should match
+    ASSERT_EQ(ovv1, ov1)
+        << "OffsetView unmanaged construction fails for rank 1";
+  }
+
+  {
+    Kokkos::Array<int64_t, 2> begins2{2, 3};
+    Kokkos::Array<int64_t, 2> ends2{5, 7};
+    Kokkos::Experimental::OffsetView<Scalar**, Device> ov2(&s, begins2, ends2);
+
+    Kokkos::View<Scalar**, Device> v2(&s, ends2[0] - begins2[0],
+                                      ends2[1] - begins2[1]);
+    Kokkos::Experimental::OffsetView<Scalar**, Device> ovv2(v2, begins2);
+
+    ASSERT_EQ(ovv2, ov2)
+        << "OffsetView unmanaged construction fails for rank 2";
+  }
+
+  {
+    Kokkos::Array<int64_t, 3> begins3{2, 3, 5};
+    Kokkos::Array<int64_t, 3> ends3{7, 11, 13};
+    Kokkos::Experimental::OffsetView<Scalar***, Device> ovv3(&s, begins3,
+                                                             ends3);
+
+    Kokkos::View<Scalar***, Device> v3(&s, ends3[0] - begins3[0],
+                                       ends3[1] - begins3[1],
+                                       ends3[2] - begins3[2]);
+    Kokkos::Experimental::OffsetView<Scalar***, Device> ov3(v3, begins3);
+
+    ASSERT_EQ(ovv3, ov3)
+        << "OffsetView unmanaged construction fails for rank 3";
+  }
+
+#ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST
+  {
+    using offset_view_type = Kokkos::Experimental::OffsetView<Scalar*, Device>;
+
+    // Range calculations must be positive
+    ASSERT_NO_THROW({ offset_view_type(&s, {0}, {1}); });
+    ASSERT_ANY_THROW({ offset_view_type(&s, {0}, {0}); });
+    ASSERT_ANY_THROW({ offset_view_type(&s, {0}, {-1}); });
```

Comment:
The second curly braces were because I had more than one statement inside.  I had to move out the using declaration as it wouldn't compile (I'm guessing because it has template parameters which don't interact well with the macro).